tests/provider: EC2 Traffic Mirror: query returned no results

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

These tests fails on GovCloud:

```

TestAccAWSEc2TrafficMirrorSession_basic
TestAccAWSEc2TrafficMirrorSession_disappears
TestAccAWSEc2TrafficMirrorSession_tags
TestAccAWSEc2TrafficMirrorTarget_eni


```

With this error: `Your query returned no results. Please change your search criteria and try again.`.

### New or Affected Resource(s)

<!--- Please list the new or affected resources and data sources. --->

* aws_ec2_traffic_mirror_target
* aws_ec2_traffic_mirror_session

### Terraform Configuration Files

Example config:

```hcl
resource "aws_instance" "src" {
  ami           = data.aws_ami.amzn-ami-minimal-hvm-ebs.id
  instance_type = "m5.large" # m5.large required because only Nitro instances support mirroring
  subnet_id     = aws_subnet.sub1.id

  tags = {
    Name = "yak mirror"
  }
}

resource "aws_lb" "lb" {
  name               = "yak-mirror"
  internal           = true
  load_balancer_type = "network"
  subnets            = [aws_subnet.sub1.id, aws_subnet.sub2.id]

  enable_deletion_protection = false

  tags = {
    Name        = "yak mirror"
    Environment = "production"
  }
}

resource "aws_ec2_traffic_mirror_filter" "filter" {
}

resource "aws_ec2_traffic_mirror_target" "target" {
  network_load_balancer_arn = aws_lb.lb.arn
}
```

### Debug Output

```
estAccAWSEc2TrafficMirrorSession_basic: resource_aws_ec2_traffic_mirror_session_test.go:27: Step 1/4 error: Error running pre-apply refresh: 2020/10/27 09:21:37 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
        
        Error: Your query returned no results. Please change your search criteria and try again.

--- FAIL: TestAccAWSEc2TrafficMirrorSession_basic (2.48s)
```

### References

* #15865

--- a/aws/resource_aws_ec2_traffic_mirror_session_test.go
+++ b/aws/resource_aws_ec2_traffic_mirror_session_test.go
@@ -183,7 +183,7 @@ func testAccCheckAWSEc2TrafficMirrorSessionExists(name string, session *ec2.Traf
 }
 
 func testAccTrafficMirrorSessionConfigBase(rName string) string {
-	return fmt.Sprintf(`
+	return composeConfig(testAccLatestAmazonLinuxHvmEbsAmiConfig(), fmt.Sprintf(`
 data "aws_availability_zones" "azs" {
   state = "available"
 
@@ -191,22 +191,6 @@ data "aws_availability_zones" "azs" {
     name   = "opt-in-status"
     values = ["opt-in-not-required"]
   }
-}
-
-data "aws_ami" "amzn-linux" {
-  most_recent = true
-
-  filter {
-    name   = "name"
-    values = ["amzn2-ami-hvm-2.0*"]
-  }
-
-  filter {
-    name   = "architecture"
-    values = ["x86_64"]
-  }
-
-  owners = ["137112412989"]
 }
 
 resource "aws_vpc" "vpc" {
@@ -238,7 +222,7 @@ resource "aws_subnet" "sub2" {
 }
 
 resource "aws_instance" "src" {
-  ami           = data.aws_ami.amzn-linux.id
+  ami           = data.aws_ami.amzn-ami-minimal-hvm-ebs.id
   instance_type = "m5.large" # m5.large required because only Nitro instances support mirroring
   subnet_id     = aws_subnet.sub1.id
 
@@ -267,22 +251,22 @@ resource "aws_ec2_traffic_mirror_filter" "filter" {
 resource "aws_ec2_traffic_mirror_target" "target" {
   network_load_balancer_arn = aws_lb.lb.arn
 }
-`, rName)
+`, rName))
 }
 
 func testAccTrafficMirrorSessionConfig(rName string, session int) string {
-	return testAccTrafficMirrorSessionConfigBase(rName) + fmt.Sprintf(`
+	return composeConfig(testAccTrafficMirrorSessionConfigBase(rName), fmt.Sprintf(`
 resource "aws_ec2_traffic_mirror_session" "test" {
   traffic_mirror_filter_id = aws_ec2_traffic_mirror_filter.filter.id
   traffic_mirror_target_id = aws_ec2_traffic_mirror_target.target.id
   network_interface_id     = aws_instance.src.primary_network_interface_id
   session_number           = %d
 }
-`, session)
+`, session))
 }
 
 func testAccTrafficMirrorSessionConfigTags1(rName, tagKey1, tagValue1 string, session int) string {
-	return testAccTrafficMirrorSessionConfigBase(rName) + fmt.Sprintf(`
+	return composeConfig(testAccTrafficMirrorSessionConfigBase(rName), fmt.Sprintf(`
 resource "aws_ec2_traffic_mirror_session" "test" {
   traffic_mirror_filter_id = aws_ec2_traffic_mirror_filter.filter.id
   traffic_mirror_target_id = aws_ec2_traffic_mirror_target.target.id
@@ -293,11 +277,11 @@ resource "aws_ec2_traffic_mirror_session" "test" {
     %[1]q = %[2]q
   }
 }
-`, tagKey1, tagValue1, session)
+`, tagKey1, tagValue1, session))
 }
 
 func testAccTrafficMirrorSessionConfigTags2(rName, tagKey1, tagValue1, tagKey2, tagValue2 string, session int) string {
-	return testAccTrafficMirrorSessionConfigBase(rName) + fmt.Sprintf(`
+	return composeConfig(testAccTrafficMirrorSessionConfigBase(rName), fmt.Sprintf(`
 resource "aws_ec2_traffic_mirror_session" "test" {
   traffic_mirror_filter_id = aws_ec2_traffic_mirror_filter.filter.id
   traffic_mirror_target_id = aws_ec2_traffic_mirror_target.target.id
@@ -309,13 +293,11 @@ resource "aws_ec2_traffic_mirror_session" "test" {
     %[3]q = %[4]q
   }
 }
-`, tagKey1, tagValue1, tagKey2, tagValue2, session)
+`, tagKey1, tagValue1, tagKey2, tagValue2, session))
 }
 
 func testAccTrafficMirrorSessionConfigWithOptionals(description string, rName string, session, pLen, vni int) string {
-	return fmt.Sprintf(`
-%s
-
+	return composeConfig(testAccTrafficMirrorSessionConfigBase(rName), fmt.Sprintf(`
 resource "aws_ec2_traffic_mirror_session" "test" {
   description              = "%s"
   traffic_mirror_filter_id = aws_ec2_traffic_mirror_filter.filter.id
@@ -325,7 +307,7 @@ resource "aws_ec2_traffic_mirror_session" "test" {
   packet_length            = %d
   virtual_network_id       = %d
 }
-`, testAccTrafficMirrorSessionConfigBase(rName), description, session, pLen, vni)
+`, description, session, pLen, vni))
 }
 
 func testAccPreCheckAWSEc2TrafficMirrorSession(t *testing.T) {

--- a/aws/resource_aws_ec2_traffic_mirror_target_test.go
+++ b/aws/resource_aws_ec2_traffic_mirror_target_test.go
@@ -226,7 +226,7 @@ resource "aws_subnet" "sub2" {
 }
 
 func testAccTrafficMirrorTargetConfigNlb(rName, description string) string {
-	return testAccTrafficMirrorTargetConfigBase(rName) + fmt.Sprintf(`
+	return composeConfig(testAccTrafficMirrorTargetConfigBase(rName), fmt.Sprintf(`
 resource "aws_lb" "lb" {
   name               = %[1]q
   internal           = true
@@ -245,29 +245,16 @@ resource "aws_ec2_traffic_mirror_target" "test" {
   description               = %[2]q
   network_load_balancer_arn = aws_lb.lb.arn
 }
-`, rName, description)
+`, rName, description))
 }
 
 func testAccTrafficMirrorTargetConfigEni(rName, description string) string {
-	return testAccTrafficMirrorTargetConfigBase(rName) + fmt.Sprintf(`
-data "aws_ami" "amzn-linux" {
-  most_recent = true
-
-  filter {
-    name   = "name"
-    values = ["amzn2-ami-hvm-2.0*"]
-  }
-
-  filter {
-    name   = "architecture"
-    values = ["x86_64"]
-  }
-
-  owners = ["137112412989"]
-}
-
+	return composeConfig(
+		testAccTrafficMirrorTargetConfigBase(rName),
+		testAccLatestAmazonLinuxHvmEbsAmiConfig(),
+		fmt.Sprintf(`
 resource "aws_instance" "src" {
-  ami           = data.aws_ami.amzn-linux.id
+  ami           = data.aws_ami.amzn-ami-minimal-hvm-ebs.id
   instance_type = "t2.micro"
   subnet_id     = aws_subnet.sub1.id
 
@@ -280,11 +267,11 @@ resource "aws_ec2_traffic_mirror_target" "test" {
   description          = %[2]q
   network_interface_id = aws_instance.src.primary_network_interface_id
 }
-`, rName, description)
+`, rName, description))
 }
 
 func testAccTrafficMirrorTargetConfigTags1(rName, description, tagKey1, tagValue1 string) string {
-	return testAccTrafficMirrorTargetConfigBase(rName) + fmt.Sprintf(`
+	return composeConfig(testAccTrafficMirrorTargetConfigBase(rName), fmt.Sprintf(`
 resource "aws_lb" "lb" {
   name               = %[1]q
   internal           = true
@@ -307,11 +294,11 @@ resource "aws_ec2_traffic_mirror_target" "test" {
     %[3]q = %[4]q
   }
 }
-`, rName, description, tagKey1, tagValue1)
+`, rName, description, tagKey1, tagValue1))
 }
 
 func testAccTrafficMirrorTargetConfigTags2(rName, description, tagKey1, tagValue1, tagKey2, tagValue2 string) string {
-	return testAccTrafficMirrorTargetConfigBase(rName) + fmt.Sprintf(`
+	return composeConfig(testAccTrafficMirrorTargetConfigBase(rName), fmt.Sprintf(`
 resource "aws_lb" "lb" {
   name               = %[1]q
   internal           = true
@@ -335,7 +322,7 @@ resource "aws_ec2_traffic_mirror_target" "test" {
     %[5]q = %[6]q
   }
 }
-`, rName, description, tagKey1, tagValue1, tagKey2, tagValue2)
+`, rName, description, tagKey1, tagValue1, tagKey2, tagValue2))
 }
 
 func testAccPreCheckAWSEc2TrafficMirrorTarget(t *testing.T) {

